apps/sync/js/adapters/bookmarks.js

```diff
@@ -167,12 +182,34 @@ var BookmarksHelper = (() => {
     });
   }
 
+  function handleClear() {
+    return getLastUserid().then(lastUserid => {
+      if (lastUserid === null) {
+        return Promise.resolve();
+      }
+      return _ensureStore().then(store => {
+        // If DataStore is empty this could be because TV logout cleared it,
+        // so unmark mTime for the previous user:
+        return store.getLength().then(numRecords => {
```

Comment:
Unfortunately, this is more complex than that. It is possible that we added some records to the DataStore *after* clearing it while the Sync app was closed. In that case we won't be removing mTime.

Instead of checking the size, we need to get the list of DataStore operations looking for a clear operation. When we create mTime, we need to also save the last clear operation timestamp, so we don't clear mTime everytime we open the Sync app.